skopt/space.py

```diff
@@ -246,13 +264,23 @@ def __init__(self, categories, prior=None):
         self.categories = categories
         self.transformer = _CategoricalEncoder()
         self.transformer.fit(self.categories)
+        self._prior = prior
```

Comment:
should arguments passed to the constructor always be assigned to `self` without the `_` for dimensions? any reason why `min` and `max` have the `_`?